src/matching_market.sol

```diff
@@ -486,126 +436,51 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
-        uint pos,          //position id
-        bool rounding      //match "close enough" orders?
-    )
-        internal
-        returns (uint id)
-    {
-        uint best_maker_id;    //highest maker id
-        uint t_buy_amt_old;    //taker buy how much saved
-        uint m_buy_amt;        //maker offer wants to buy this much token
-        uint m_pay_amt;        //maker offer wants to sell this much token
-
-        // there is at least one offer stored for token pair
-        while (_best[t_buy_gem][t_pay_gem] > 0) {
-            best_maker_id = _best[t_buy_gem][t_pay_gem];
-            m_buy_amt = offers[best_maker_id].buy_amt;
-            m_pay_amt = offers[best_maker_id].pay_amt;
-
-            // Ugly hack to work around rounding errors. Based on the idea that
-            // the furthest the amounts can stray from their "true" values is 1.
-            // Ergo the worst case has t_pay_amt and m_pay_amt at +1 away from
-            // their "correct" values and m_buy_amt and t_buy_amt at -1.
-            // Since (c - 1) * (d - 1) > (a + 1) * (b + 1) is equivalent to
-            // c * d > a * b + a + b + c + d, we write...
-            if (mul(m_buy_amt, t_buy_amt) > mul(t_pay_amt, m_pay_amt) +
-                (rounding ? m_buy_amt + t_buy_amt + t_pay_amt + m_pay_amt : 0))
-            {
-                break;
-            }
-            // ^ The `rounding` parameter is a compromise borne of a couple days
-            // of discussion.
-            buy(best_maker_id, min(m_pay_amt, t_buy_amt));
-            t_buy_amt_old = t_buy_amt;
-            t_buy_amt = sub(t_buy_amt, min(m_pay_amt, t_buy_amt));
-            t_pay_amt = mul(t_buy_amt, t_pay_amt) / t_buy_amt_old;
-
-            if (t_pay_amt == 0 || t_buy_amt == 0) {
-                break;
-            }
-        }
-
-        if (t_buy_amt > 0 && t_pay_amt > 0 && t_pay_amt >= _dust[t_pay_gem]) {
-            //new offer should be created
-            id = super.offer(t_pay_amt, t_pay_gem, t_buy_amt, t_buy_gem);
-            //insert offer into the sorted list
-            _sort(id, pos);
-        }
-    }
-
-    // Make a new offer without putting it in the sorted list.
-    // Takes funds from the caller into market escrow.
-    // ****Available to authorized contracts only!**********
-    // Keepers should call insert(id,pos) to put offer in the sorted list.
-    function _offeru(
-        uint pay_amt,      //maker (ask) sell how much
-        ERC20 pay_gem,     //maker (ask) sell which token
-        uint buy_amt,      //maker (ask) buy how much
-        ERC20 buy_gem      //maker (ask) buy which token
-    )
-        internal
-        returns (uint id)
-    {
-        require(_dust[pay_gem] <= pay_amt);
-        id = super.offer(pay_amt, pay_gem, buy_amt, buy_gem);
-        _near[id] = _head;
-        _head = id;
-        LogUnsortedOffer(id);
-    }
-
-    //put offer into the sorted list
+    // Put offer into the sorted list
```

Comment:
It would be really helpful to state somewhere that the best offer (which is defined as highest ratio of `pay_amt` to `buy_amt`) is the tail of the list. Maybe it fits better up with the variable definitions, but having that knowledge would make it faster/easier to decipher the code.